gui/src/MainWindow.h

```diff
@@ -0,0 +1,178 @@
+/********************************************************************************
+* Copyright 2017 DigitalGlobe, Inc.
+* Author: Kevin McGee
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+********************************************************************************/
+#ifndef MAINWINDOW_H
+#define MAINWINDOW_H
+
+#include <string>
+#include <iostream>
+#include <sstream>
+
+#include <boost/make_unique.hpp>
+
+#include <opencv2/core/types.hpp>
+
+#include <QMainWindow>
+#include <QDebug>
+#include <QFileDialog>
+#include <QTextStream>
+#include <QMessageBox>
+#include <QValidator>
+#include <QCloseEvent>
+#include <QStatusBar>
+#include <QProgressBar>
+
+#include "OSNModelThread.h"
+#include <OpenSpaceNetArgs.h>
+#include <OpenSpaceNet.h>
+#include <OsnProcessingThread.h>
+#include "OSNProgressDisplay.h"
+#include <imagery/MapServiceClient.h>
+
+#include "ProgressWindow.h"
+#include "QDebugStream.h"
```

Comment:
All of these includes need to be examined to determine if they are needed here or in the cpp.  At the very least `<boost/make_unique.hpp>` does not belong here and `cv::Size` can be forward-declared here and then included in the .cpp file.

I am not sure why `boost::shared_ptr` is preferred over `std::shared_ptr` here or anywhere else in the code but you are also missing the include for it.

Since streams are being included in the .h file you can use `<iosfwd>` here and the actual includes (`<iostream>`, `<sstream>`, etc) in the .cpp file.  This rule should apply to this header and every other header in OSN.